lib/WeBWorK3/Mojolicious.pm

```diff
@@ -16,7 +16,7 @@ our $exception_handler = sub {
 			$next->();
 		} catch {
 			my $output = {exception => ref($_)};
-			$output->{message} = $_->message if ($_ && ref($_) eq 'Mojo::Exception' or ref($_) =~ /^DB::Exception/x);
+			$output->{message} = $_->message if ($_ and ref($_) eq 'Mojo::Exception' or ref($_) =~ /^DB::Exception/x);
```

Comment:
Note that "a && b or c" is logically equivalent to "a and b or c" (these are also logically equivalent to "a && b || c").  So in terms of what is desired, this doesn't change anything.  Of course perl critic complains about this with the policy ValuesAndExpressions::ProhibitMixedBooleanOperators set at its defaults and severity level 4, which is probably why you made this change.  (Further note that "a and b || c" is logically different than "a and b or c" and the others.)

However, both before and after this change this is logically incorrect.  Parentheses are necessary in this situation around the "or" part.  With the way it was before and with this change if `$_` is falsy, then the last condition will still be checked, and that should not be the case here.  (In fact "a and b || c" does work correctly here without parentheses, although it breaks the perlcritic policy.)

Furthermore, you made the wrong change to satisfy the no mixed boolean operator policy in my opinion.  You changed the `&&` to `and`, and you should have changed `or` to `||`.  Generally speaking always prefer `&&` and `||` to `and` and `or` unless it is one of the very few specific situations where the latter should be used (like for control flow).

So in short the correct (as well as opinionated) change should be:
```perl
$output->{message} = $_->message if ($_ && (ref($_) eq 'Mojo::Exception' || ref($_) =~ /^DB::Exception/x));
```

Reply:
I think these are changes from #36 -- I don't recall making this change.

Reply:
Also, agreed -- && > 'and', || > 'or'

Reply:
I see. That is what I get for commenting before that was merged. I can't sort out what is this pull request and what is that.

Reply:
I think I covered all the 'changed files' in the description -- but yeah, it will be much easier to review once the previous PR gets merged...

Reply:
Oh, I was supposed to read that!